server/plugins/FilterUGens.cpp

```diff
@@ -1053,7 +1053,15 @@ void Integrator_Ctor(Integrator* unit) {
         SETCALC(Integrator_next);
     unit->m_b1 = ZIN0(1);
     unit->m_y1 = 0.f;
-    Integrator_next(unit, 1);
+    // Note: Here, we deliberately do NOT want to call Integrator_next(unit, 1).
+    // That would initialize the presample AND update m_y1.
+    // Then this initial value will be added in *again* by the normal calc loop.
+    // So the integral would be corrupted for its lifetime.
+    // The formula is input + (coef * output[-1]).
+    // But output[-1] is 0 initially, so we just drop that term.
+    float* out = ZOUT(0);
+    float* in = ZIN(0);
+    ZXP(out) = ZXP(in);
```

Comment:
Could this not be resolved by again setting `unit->m_y1  = 0` after calling `Integrator_next(unit, 1);`?
I previously had this changed staged for a batch of init sample fixes:
```
void Integrator_Ctor(Integrator* unit) {
    if (INRATE(1) == calc_ScalarRate)
        SETCALC(Integrator_next_i);
    else
        SETCALC(Integrator_next);
    unit->m_b1 = ZIN0(1);
    unit->m_y1 = 0.f;
    Integrator_next(unit, 1); 
    unit->m_y1 = 0.f;
}
```
I had tested a few cases at both a- and k- rate, but I can't  retest this atm, but perhaps you  could patch this in and see if it gives the same result? (great that you have a unit test for it!)

Reply:
Also, regarding the comment—the problem you're describing is quite pervasive so describing each fix in such detail will really crowd the source. I would hope that once more people recognize the proper initialization pattern it will be more obvious why it's structured that way. Or just noting briefly something like:
```
...
Integrator_next(unit, 1); 
unit->m_y1 = 0.f; // reset initial state
...
```

Reply:
> Could this not be resolved by again setting unit->m_y1 = 0 after calling Integrator_next(unit, 1);?

That seems inefficient to me, though. I don't see that Integrator_next is accomplishing anything at initialization time that is not accomplished by simply initializing the two member variables.

I can see that in most filters, you need a `_next` function to calculate the initial coefficients. That isn't the case here. My preference would be to eliminate redundant operations.

> Also, regarding the comment—the problem you're describing is quite pervasive so describing each fix in such detail will really crowd the source.

That's a fair point. The text could be shorter.

I question the notion that every UGen must call `_next(unit, 1)` in the Ctor. It's a useful rule of thumb, but its application-by-default is the cause of some of the initialization bugs that you're working on. It should be considered case-by-case. Exceptional cases probably should be noted with a comment.

Reply:
> I question the notion that every UGen must call `_next(unit, 1)` in the Ctor. It's a useful rule of thumb, but its application-by-default is the cause of some of the initialization bugs that you're working on. 
I've thought through this in the past I typically landed on the side of sticking to a pattern used throughout the codebase and set by the original author (this was informed a long time ago by a reviewer). I have the same hesitation about wasted computation and rationalized it by considering that this is only one frame of calculation, but I have, in some cases, moved calculations into the `ctor` in the interest of efficiency and/or clariy. So I see the case for both.

> Exceptional cases probably should be noted with a comment.
I agree. In the case of setting the output sample directly, I think it could be a good idea to note what the calculation "formula" is in a comment, in a way indicating that the init sample == first sample concern  has been considered. Something like
```
ZXP(out) = ZXP(in); // out = in + (coef * out[-1]); 
```

Just thinking out loud because this discussion will inform how I continue to handle this type of bug.